src/main/java/edu/njit/jerse/ashe/utils/JavaCodeParser.java

```diff
@@ -206,23 +245,35 @@ public static String extractMethodName(String targetMethod) {
      * Extracts the body of a specified method from the given Java code string.
      *
      * @param method the entire Java method as a string
-     * @return the body of the method as a string, or an empty string if not found
+     * @return the body of the method as a string
+     * @throws NoSuchElementException if the method declaration or body is not found
      */
-    public static String extractMethodBody(String method) {
+    public static String extractMethodBody(String method) throws NoSuchElementException {
```

Comment:
This method can also throw a `ParseProblemException`, but that isn't documented.